sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -444,12 +445,22 @@ class AggHelper(
   } else {
     inputAttributes
   }
-  val preStepBound = GpuBindReferences.bindGpuReferencesTiered(preStep.toList,
-    preStepAttributes.toList, conf)
+  val preStepBound = metrics match {
+    case Some(m) =>
+      GpuBindReferences.bindGpuReferencesTiered(preStep.toList, preStepAttributes.toList, conf, m)
```

Comment:
if we made a bindGpuReferencesTiered with an optional metrics map, we could remove some of this. I haven't looked further in the PR, but I assume there's some repetition here.